src/forms/StripeConnectAccountForm/StripeConnectAccountForm.js

```diff
@@ -0,0 +1,278 @@
+          handleSubmit,
+          inProgress,
+          intl,
+          invalid,
+          pristine,
+          ready,
+          savedCountry,
+          stripeAccountFetched,
+          stripeBankAccountLastDigits,
+          submitButtonText,
+          values,
+          stripeConnected,
+        } = fieldRenderProps;
+
+        const accountDataLoaded = stripeConnected && (stripeAccountFetched && savedCountry);
+        const submitInProgress = inProgress;
+        const submitDisabled = pristine || invalid || disabled || submitInProgress;
+
+        const handleFormSubmit = event => {
+          // Close the bank account form when clicking "save details"
+          setShowCardUpdateInput(false);
+          handleSubmit(event);
+        };
+
+        const countryLabel = intl.formatMessage({ id: 'StripeConnectAccountForm.countryLabel' });
+        const classes = classNames(css.root, className, {
+          [css.disabled]: disabled,
+        });
+
+        return config.stripe.publishableKey ? (
+          <Form className={classes} onSubmit={handleFormSubmit}>
+            {!stripeConnected ? (
+              <CreateStripeAccountFields
+                stripeConnected={stripeConnected}
+                disabled={disabled}
+                countryLabel={countryLabel}
+                supportedCountries={supportedCountries}
+                values={values}
+                intl={intl}
+              />
+            ) : accountDataLoaded ? (
+              <UpdateStripeAccountFields
+                disabled={disabled}
+                countryLabel={countryLabel}
+                savedCountry={savedCountry}
+                stripeBankAccountLastDigits={stripeBankAccountLastDigits}
+                showCardUpdateInput={showCardUpdateInput}
+                values={values}
+                submitInProgress={submitInProgress}
+                setShowCardUpdateInput={setShowCardUpdateInput}
+                intl={intl}
+              />
+            ) : (
+              <div className={css.savedInformation}>
+                <FormattedMessage id="StripeConnectAccountForm.loadingStripeAccountData" />
+              </div>
+            )}
+
+            <ErrorsMaybe stripeAccountError={stripeAccountError} />
+            {children}
+
+            {!stripeConnected || accountDataLoaded ? (
+              <Button
+                className={css.submitButton}
+                type="submit"
+                inProgress={submitInProgress}
+                disabled={submitDisabled}
+                ready={ready}
+              >
+                {submitButtonText || (
+                  <FormattedMessage id="StripeConnectAccountForm.submitButtonText" />
+                )}
+              </Button>
+            ) : null}
+          </Form>
+        ) : (
+          <div className={css.missingStripeKey}>
+            <FormattedMessage id="StripeConnectAccountForm.missingStripeKey" />
+          </div>
+        );
+      }}
+    />
+  );
```

Comment:
This whole return statement above has so much parallel and nested ternaries that it's really hard to fully understand. Could it be split down to variables above?